activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb

```diff
@@ -38,14 +38,21 @@ def visit_AddColumnDefinition(o)
           end
 
           def visit_TableDefinition(o)
-            create_sql = "CREATE#{' TEMPORARY' if o.temporary} TABLE "
-            create_sql << "#{quote_table_name(o.name)} "
-            create_sql << "(#{o.columns.map { |c| accept c }.join(', ')}) " unless o.as
+            create_sql = "CREATE#{' TEMPORARY' if o.temporary} TABLE #{quote_table_name(o.name)} "
+
+            statements = o.columns.map { |c| accept c }
+            statements << accept(o.primary_keys) if o.primary_keys
+
+            create_sql << "(#{statements.join(', ')}) " if statements.present?
```

Comment:
This is for MySQL. MySQL creates new columns for all elements in the SELECT. For example:

```
mysql> CREATE TABLE test (a INT NOT NULL AUTO_INCREMENT,
    ->        PRIMARY KEY (a), KEY(b))
    ->        ENGINE=MyISAM SELECT b,c FROM test2;
```
http://dev.mysql.com/doc/refman/5.6/en/create-table-select.html

I would like to eliminate the duplication code of `visit_TableDefinition` by #21589.
This change needs for MySQL. And not broken existing code.